server/controllers/homepageController.ts

```diff
@@ -1,34 +1,53 @@
 import { NextFunction, Request, RequestHandler, Response } from 'express'
-import { userHasRoles } from '../utils/utils'
 import { Role } from '../enums/role'
 import config from '../config'
+import { CustomRequest } from '../data/interfaces/customRequest'
+import { WhereAboutsConfig } from '../data/interfaces/whereAboutsConfig'
+import { KeyWorkerPrisonStatus } from '../data/interfaces/keyWorkerPrisonStatus'
+import { ServiceSubLink } from '../data/interfaces/serviceSubLink'
 import HomepageService from '../services/homepageService'
 import HmppsCache from '../middleware/hmppsCache'
 
+// eslint-disable-next-line import/prefer-default-export
+export const hasAnyRole = (requiredRoles: string[], userRoles: string[]): boolean =>
+  requiredRoles.some(role => userRoles.includes(role))
+
 /**
  * Parse requests for case notes routes and orchestrate response
  */
 export default class HomepageController {
   constructor(private readonly homepageService: HomepageService, private readonly todayCache: HmppsCache) {}
 
   public displayHomepage(): RequestHandler {
-    return async (req: Request, res: Response, next: NextFunction) => {
-      const { userRoles, activeCaseLoadId } = res.locals.user
+    return async (req: Request & CustomRequest, res: Response, next: NextFunction) => {
+      const { activeCaseLoadId } = res.locals.user
 
       // Search Section
       const errors = req.flash('errors')
-      const userHasGlobal = userHasRoles([Role.GlobalSearch], userRoles)
+      const userHasRoles = (roles: string[]) => hasAnyRole(res.locals.user.userRoles, roles)
```

Comment:
Seems confusing to create your own function with the same name as the existing function from `utils.ts`
I think it's better to call your new function something else.